### PR TITLE
AС-1 > Add first screen with Bloc

### PR DESCRIPTION
**Pull request by yarshaqa**
bloc is ready

### CODE REVIEW

**Comment by VovaMedvediev** [comment]
Add updated .gitignore (send in DM)

**Comment by VovaMedvediev** [comment]
Add updated linter rules (send in DM)

and add following lines to pubspec.yaml:

dev_dependencies:
  flutter_test:
    sdk: flutter

  # RULES:
  lint: ^1.8.1

File: lib/main.dart [hunk 1]
```diff
@@ -27,11 +29,8 @@ class _HomePageState extends State<HomePage> {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
-        appBar: AppBar(leading: const Text('Age Calculator')),
```
Comment (VovaMedvediev) [inline lib/main.dart:30]: Extract HomePage widget to separate file. 
It is better to have the root widget of your app in main.dart and other widgets in separate files.

File: lib/main.dart [hunk 2]
```diff
@@ -27,11 +29,8 @@ class _HomePageState extends State<HomePage> {
...
-      body: const Center(
```
Comment (VovaMedvediev) [inline lib/main.dart:31]: For what reason do you need HomePage as StatefulWidget? I don't see any manipulations with State so it's better to refactor to StatelessWidget.

File: lib/input_data_page.dart [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+              }
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+
+  Widget datePick(String widgetName, DateTime pick) {
```
Comment (VovaMedvediev) [inline lib/input_data_page.dart]: Extract to separate widget

Comment (VovaMedvediev) [reply lib/input_data_page.dart]: Why?
[Click here](https://stackoverflow.com/questions/53234825/what-is-the-difference-between-functions-and-classes-to-create-reusable-widgets)

File: lib/input_data_page.dart [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+                        : pick.toString())),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<DateTime?> showPicker(String widgetName) {
```
Comment (VovaMedvediev) [inline lib/input_data_page.dart]: Move to above-mentioned new widget.
Let this function be private

File: lib/constants.dart [hunk 5]
```diff
@@ -0,0 +1,6 @@
+class Constants {
```
Comment (VovaMedvediev) [inline lib/constants.dart]: I think class Constants is redundant.
You can write "const int defaultDate = 0000;"

File: lib/constants.dart [hunk 6]
```diff
@@ -0,0 +1,6 @@
...
+  static const int defaultDate = 0000;
+  static const int standartHeight = 15;
```
Comment (VovaMedvediev) [inline lib/constants.dart]: Fix typo

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

**Comment by VovaMedvediev** [comment]
Also it will be cool to prepare such directories structure:
main.dart in the lib directory
screens in the lib/screens/ directory
for example - HomePage Widget in lib/screens/home_page
InputDataPage in lib/screens/input_data_page
etc

**Comment by VovaMedvediev** [comment]
And fix all linter warning please :)

File: lib/main.dart (hunk 1 above)
Comment (yarshaqa) [reply lib/main.dart:30]: moved to separate file

File: lib/main.dart (hunk 2 above)
Comment (yarshaqa) [reply lib/main.dart:31]: Refactored HomePage class to stateless widget

File: lib/input_data_page.dart (hunk 3 above)
Comment (yarshaqa) [reply lib/input_data_page.dart]: Extracted

File: lib/input_data_page.dart (hunk 4 above)
Comment (yarshaqa) [reply lib/input_data_page.dart]: showPicker moved to DatePick class and marked private
on raw #56 changed to show just SnackBar

File: lib/constants.dart (hunk 5 above)
Comment (yarshaqa) [reply lib/constants.dart]: I planned in this project to use the Constants class for all possible constants
why is it redundant? Not good practice?

File: lib/constants.dart (hunk 6 above)
Comment (yarshaqa) [reply lib/constants.dart]: fixed

File: lib/constants.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/constants.dart]: sure, it is non-recommended practice.
[AVOID defining a class that contains only static members](https://dart.dev/guides/language/effective-dart/design#avoid-defining-a-class-that-contains-only-static-members)

File: lib/input_data_page.dart [hunk 7]
```diff
@@ -0,0 +1,141 @@
+import 'calculated_data_page.dart';
+import 'constants.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'input_data_bloc.dart';
+
+class InputDataPage extends StatefulWidget {
```
Comment (VovaMedvediev) [inline lib/input_data_page.dart]: For what reason do you need InputDataPage as Stateful Widget?
Maybe would you like to create your Bloc in initState?

File: lib/input_data_page.dart [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+            },
+            child: const Text('Calculate Age'),
+          )
+        ],
+      ),
+    );
+  }
+}
+
+class DatePick extends StatelessWidget {
```
Comment (VovaMedvediev) [inline lib/input_data_page.dart]: Extract to separate file

File: lib/input_data_page.dart [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+                        : pick.toString())),
+              ],
+            ),
+          ),
+        ),
+      ],
+    );
+  }
+
+  Future<DateTime?> _showPicker(BuildContext context, String widgetName) {
```
Comment (VovaMedvediev) [inline lib/input_data_page.dart]: extract to above mentioned separate file

**Review by VovaMedvediev — CHANGES_REQUESTED** [review]

File: lib/constants.dart (hunk 5 above)
Comment (yarshaqa) [reply lib/constants.dart]: made import as constants

File: lib/input_data_page.dart (hunk 7 above)
Comment (yarshaqa) [reply lib/input_data_page.dart]: Refactored to Stateless widget

File: lib/input_data_page.dart (hunk 8 above)
Comment (yarshaqa) [reply lib/input_data_page.dart]: extracted DatePick class to date_pick_widget.dart file

File: lib/input_data_page.dart (hunk 9 above)
Comment (yarshaqa) [reply lib/input_data_page.dart]: extracted to separate date_pick_widget.dart file

**Review by VovaMedvediev — APPROVED** [review]

**Pull request merged**